core/src/main/java/org/apache/iceberg/encryption/Ciphers.java

```diff
@@ -53,10 +63,14 @@ public AesGcmEncryptor(byte[] keyBytes) {
       this.randomGenerator = new SecureRandom();
     }
 
-    public byte[] encrypt(byte[] plainText, byte[] aad) {
+    public byte[] encrypt(byte[] plaintext, byte[] aad) {
+      return encrypt(plaintext, 0, plaintext.length, aad);
+    }
+
+    public byte[] encrypt(byte[] plaintext, int plaintextOffset, int plaintextLength, byte[] aad) {
```

Comment:
Where is this encryption format documented so I can take a look at compare it to the spec?

Reply:
Yep, this (and other GCM Stream fields) are not documented yet. In a way, the GCM Streams are a simple file format, designed to add encryption and integrity verification to existing Input/Output Streams, in a transparent (and efficient) manner. I agree we need an explicit spec for this. Should I add a .md file (to the `iceberg/format` folder) in this PR so we can discuss it here?

Reply:
@rdblue I've created a new PR, with the spec doc: https://github.com/apache/iceberg/pull/5432